qtism/data/content/interactions/GapMatchInteraction.php

```diff
@@ -188,6 +190,27 @@ public function getComponents()
         return new QtiComponentCollection(array_merge($parentComponents->getArrayCopy(), $this->getGapChoices()->getArrayCopy(), $this->getContent()->getArrayCopy()));
     }
 
+    public function getResponseValidityConstraint(): ?ResponseValidityConstraint
```

Comment:
There's no null return in this function. Please check return types in other classes because other also have not needed null return.